test/collections.js

```diff
@@ -317,6 +317,10 @@
     assert.deepEqual(_.filter(list, {b: 2}), [{a: 1, b: 2}, {a: 2, b: 2}]);
     assert.deepEqual(_.filter(list, {}), list, 'Empty object accepts all items');
     assert.deepEqual(_(list).filter({}), list, 'OO-filter');
+
+    // Works with RegExps
+    var stringArray = ['cat', 'dog', 'catdog'];
+    assert.deepEqual(_.filter(stringArray, /cat/), ['cat', 'catdog']);
```

Comment:
It would probably be best to test `_.iteratee` as it's a more direct way of testing `cb()`: https://github.com/jashkenas/underscore/blob/67e5c9eb585124d7a6dda2c1012e4ac240678f6e/test/functions.js#L678-L691